native/android/SwrveSDKUnityBridge/src/main/java/com/swrve/sdk/SwrveUnityCommon.java

```diff
@@ -0,0 +1,392 @@
+
+        return text.toString();
+    }
+
+    private String getStringDetail(String key) {
+        if(currentDetails.containsKey(key)) {
+            return (String)currentDetails.get(key);
+        }
+        return null;
+    }
+
+    private int getIntDetail(String key) {
+        if(currentDetails.containsKey(key)) {
+            return ((Double)currentDetails.get(key)).intValue();
+        }
+        return 0;
+    }
+
+    @Override
+    public String getApiKey() {
+        return getStringDetail("apiKey");
+    }
+
+    @Override
+    public String getSessionKey() {
+        return this.sessionKey;
+    }
+
+    @Override
+    public short getDeviceId() {
+        if(currentDetails.containsKey("deviceId")) {
+            return ((Double)currentDetails.get("deviceId")).shortValue();
+        }
+        return 0;
+    }
+
+    @Override
+    public int getAppId() {
+        return getIntDetail("appId");
+    }
+
+    @Override
+    public String getUserId() {
+        return getStringDetail("userId");
+    }
+
+    public String getSwrvePath() {
+        return getStringDetail("swrvePath");
+    }
+
+    String getPrefabName() { return getStringDetail(PREFAB_NAME_KEY); }
+
+    @Override
+    public String getSwrveSDKVersion() {
+        return getStringDetail(SDK_VERSION_KEY);
+    }
+
+    public String getSwrveTemporaryPath() {
+        return getStringDetail(SWRVE_TEMPORARY_PATH_KEY);
+    }
+
+    public String getLocTag() {
+        return getStringDetail("locTag");
+    }
+
+    public String getSigSuffix() {
+        return getStringDetail("sigSuffix");
+    }
+
+    @Override
+    public String getAppVersion() {
+        return getStringDetail("appVersion");
+    }
+
+    @Override
+    public String getUniqueKey() {
+        return getStringDetail("uniqueKey");
+    }
+
+    @Override
+    public String getBatchURL() {
+        return getEventsServer() + getStringDetail("batchUrl");
+    }
+
+    @Override
+    public void setLocationSegmentVersion(int locationSegmentVersion) {
+        sendMessageUp("SetLocationSegmentVersion", Integer.toString(locationSegmentVersion));
+    }
+
+    @Override
+    public String getCachedData(String userId, String key) {
+        if(LOCATION_CAMPAIGN_CATEGORY.equals(key))
+        {
+            return readFile(getSwrvePath(), getLocTag() + userId);
+        }
+        return null;
+    }
+
+    public void showConversation(String conversation) {
+        try {
+            Intent intent = new Intent(context.get(), ConversationActivity.class);
+            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+            intent.putExtra("conversation", new SwrveBaseConversation(new JSONObject(conversation), cacheDir));
+            context.get().startActivity(intent);
+        }
+        catch (JSONException exc) {
+            SwrveLogger.e(LOG_TAG, "Could not JSONify conversation, conversation string didn't have the correct structure.");
+        }
+    }
+
+    public int conversationVersion() {
```

Comment:
Can't ISwrveConversationSDK.CONVERSATION_VERSION be referenced directly? or is this just a helper?